src/vs/workbench/contrib/markers/browser/markersModel.ts

```diff
@@ -117,6 +118,19 @@ export class Marker {
 	}
 }
 
+export class MarkerTableItem extends Marker {
```

Comment:
This feels weird. Wouldn't it be best if `MarkerTableItem` doesn't extend from `Marker`? Conceptually, `Marker` is a model object while `MarkerTableItem` seems like a view model object, no?

Reply:
Happy to discuss this further to see if there is an implementation that feel more natural.

Reply:
Closing this comment as @sandy081 and I have discussed this and agreed on a plan to address this.